docs/DeveloperGuide.adoc

```diff
@@ -227,20 +250,159 @@ image::CommitActivityDiagram.png[]
 
 ===== Aspect: Data structure to support the undo/redo commands
 
-* **Alternative 1 (current choice):** Use a list to store the history of address book states.
+* **Alternative 1 (current choice):** Use a list to store the history of question bank states.
 ** Pros: Easy for new Computer Science student undergraduates to understand, who are likely to be the new incoming developers of our project.
 ** Cons: Logic is duplicated twice. For example, when a new command is executed, we must remember to update both `HistoryManager` and `VersionedAddressBook`.
 * **Alternative 2:** Use `HistoryManager` for undo/redo
 ** Pros: We do not need to maintain a separate list, and just reuse what is already in the codebase.
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
 // end::undoredo[]
 
-// tag::dataencryption[]
-=== [Proposed] Data Encryption
+// tag::submit[]
+=== [Proposed] Submit feature
+
+The submit mechanism is facilitated by `UserInterface`.
+
+The Sequence Diagram given below is an example of a scenario in which the submit command is executed when the user enters "submit" into the `CommandBox` terminal.
+
+image::submitSequenceDiagram.png[]
```

Comment:
Activation bar for command parser should end once it returns.
CommandParser and Command should be SubmitCommandParser and Submit Command.
Weird box on the right, bug?
Dotted line from command should end at the cross but I think it's a plantUML bug.